tests/commands/autoupdate_test.py

```diff
@@ -263,6 +263,27 @@ def test_does_not_reformat(tmpdir, out_of_date, store):
     assert cfg.read() == expected
 
 
+def test_does_not_change_mixed_endlines(up_to_date, tmpdir, store):
+    """Test that line endings are not being changed."""
+    fmt = (
+        'repos:\n'
+        '-   repo: {}\n'
+        '    rev: {}  # definitely the version I want!\r\n'
+        '    hooks:\r\n'
+        '    -   id: foo\n'
+        '        # These args are because reasons!\r\n'
+        '        args: [foo, bar, baz]\r\n'
+    )
+    cfg = tmpdir.join(C.CONFIG_FILE)
+    expected = bytes(
+        fmt.format(up_to_date, git.head_rev(up_to_date)), 'utf-8',
+    )
+    cfg.write_binary(expected)
+
+    assert autoupdate(str(cfg), store, freeze=False, tags_only=False) == 0
+    assert cfg.read_binary() == expected
```

Comment:
this test only covers half of your patch -- it would need to trigger the write case to cover the full patch

Reply:
You are right. It's me trying to simplify test. It did test write before I tried to "clean it up"